ibm/data_source_ibm_cis_certificates.go

```diff
@@ -0,0 +1,162 @@
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "certificate identifier",
+						},
+						cisCertificateOrderID: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "certificate id",
+						},
+						cisCertificatesType: {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "certificate type",
+						},
+						cisCertificateOrderHosts: {
+							Type:        schema.TypeList,
+							Description: "Hosts which certificate need to be ordered",
+							Required:    true,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+						},
+						cisCertificateOrderStatus: {
+							Type:        schema.TypeString,
+							Description: "certificate status",
+							Computed:    true,
+						},
+						cisCertificatesPrimaryCertificate: {
+							Type:        schema.TypeString,
+							Description: "Primary certificate id",
+							Computed:    true,
+						},
+						cisCertificatesCertificates: {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "Collection of certificates associated with this certificates",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									cisCertificatesCertificatesID: {
+										Type:        schema.TypeString,
+										Description: "certificate id",
+										Computed:    true,
+									},
+									cisCertificatesCertificatesHosts: {
+										Type:        schema.TypeList,
+										Description: "Hosts which certificates are ordered",
+										Computed:    true,
+										Elem:        &schema.Schema{Type: schema.TypeString},
+									},
+									cisCertificatesCertificatesStatus: {
+										Type:        schema.TypeString,
+										Description: "certificate status",
+										Computed:    true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+func dataIBMCISCertificatesRead(d *schema.ResourceData, meta interface{}) error {
+	cisClient, err := meta.(ClientSession).CisSSLClientSession()
+	if err != nil {
+		return err
+	}
+	crn := d.Get(cisID).(string)
+	zoneID, _, _ := convertTftoCisTwoVar(d.Get(cisDomainID).(string))
+	cisClient.Crn = core.StringPtr(crn)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+	opt := cisClient.NewListCertificatesOptions()
+	result, response, err := cisClient.ListCertificates(opt)
+	if err != nil {
+		log.Printf("List all certificates failed: %v", response)
+		return err
+	}
+	certificatesList := make([]interface{}, 0)
+	for _, instance := range result.Result {
+		certificate := map[string]interface{}{}
+		certificate["id"] = convertCisToTfThreeVar(*instance.ID, zoneID, crn)
+		certificate[cisCertificateOrderID] = *instance.ID
+		certificate[cisCertificateOrderStatus] = *instance.Status
+		certificate[cisCertificatesPrimaryCertificate] =
+			convertCISCertificatesObj(*instance.Type, instance.PrimaryCertificate)
```

Comment:
why is the `convertCISCertificatesObj` function added here?

Reply:
this will list both dedicated and custom certificate. some certificate are having integer certificate id. some has string type. so converted to make it as string.